### PR TITLE
Update sdn, sdn-ovs, sync, fluentd, descheduler to have system-cluster/node-critical priority classes.

### PR DESCRIPTION
**Pull request by aveshagarwal**
@derekwaynecarr @sjenning @ravisantoshgudimetla

### CODE REVIEW

File: roles/openshift_sdn/files/sdn-ovs.yaml [hunk 1]
```diff
@@ -29,6 +29,7 @@ spec:
       serviceAccountName: sdn
       hostNetwork: true
       hostPID: true
+      priorityClassName: system-cluster-critical
```
Comment (ravisantoshgudimetla) [inline roles/openshift_sdn/files/sdn-ovs.yaml]: I think most of them should be node critical, as it it the highest priority in the cluster.

Comment (aveshagarwal) [reply roles/openshift_sdn/files/sdn-ovs.yaml]: > I think most of them should be node critical, as it it the highest priority in the cluster.

why?

Comment (ravisantoshgudimetla) [reply roles/openshift_sdn/files/sdn-ovs.yaml]: There is a chance that system-node-critical pods will evict them system-cluster-critical, rare but possible.

Comment (aveshagarwal) [reply roles/openshift_sdn/files/sdn-ovs.yaml]: I kind of agree, (though now I see there name is a bit misleading, atleast to me, it seemed like cluster-critical is higher than node-critical, but node-critical is 1000+ of cluster critical) so I think I am planning for following:

descheduler: system-cluster-critical (it is ok if it is moved by system-node-critical) 
syn, sdn,sdn-ovs: system-node-critical  (as these will be on each node irrespective of their types)
fluentd: system-cluster-critical (in my understanding, I think, it's ok to keep them at system-cluster-critical)

**Comment by aveshagarwal** [comment]
@sdodson @michaelgugino any idea why the following error:

```
+ ansible-playbook -vv --become --become-user root --connection local --inventory sjb/inventory/ -e openshift_deployment_type=origin -e debug_level=4 -e openshift_logging_install_logging=True -e openshift_logging_image_prefix=openshift/origin- -e openshift_logging_elasticsearch_proxy_image_prefix=docker.io/openshift/ -e openshift_hosted_logging_hostname=kibana.127.0.0.1.nip.io -e openshift_logging_master_public_url=https://localhost:8443 -e openshift_master_logging_public_url=https://kibana.127.0.0.1.nip.io -e openshift_logging_use_mux=True -e openshift_logging_mux_allow_external=True -e openshift_logging_es_allow_external=True -e openshift_logging_es_ops_allow_external=True -e openshift_pkg_version=-3.11.0-0.alpha.0.329.dab335c -e openshift_release=3.11 -e openshift_logging_image_version=latest -e 'oreg_url=openshift/origin-${component}:latest' -e openshift_logging_elasticsearch_proxy_image=docker.io/openshift/oauth-proxy:v1.0.0 -e 'openshift_logging_es_nodeselector={node-role.kubernetes.io/infra:' 'true}' -e 'openshift_logging_es_ops_nodeselector={node-role.kubernetes.io/infra:' 'true}' -e skip_sanity_checks=true -e 'openshift_disable_check=*' -e openshift_install_examples=false -e debug_level=2 /usr/share/ansible/openshift-ansible/playbooks/openshift-logging/config.yml --skip-tags=update_master_config
ansible-playbook 2.5.5
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/origin/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 2.7.5 (default, Feb 20 2018, 09:19:12) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]
Using /etc/ansible/ansible.cfg as config file
**ERROR! the playbook: true} could not be found**
++ export status=FAILURE
++ status=FAILURE
+ set +o xtrace
```

File: roles/openshift_sdn/files/sdn-ovs.yaml (hunk 1 above)
Comment (ravisantoshgudimetla) [reply roles/openshift_sdn/files/sdn-ovs.yaml]: I think fluentd should be at node-critical as this should not be evicted at all. If evicted, we will not have logging from the node.

Comment (aveshagarwal) [reply roles/openshift_sdn/files/sdn-ovs.yaml]: > I think fluentd should be at node-critical as this should not be evicted at all. If evicted, we will not have logging from the node.

What  do you think should have higher priority between "sdn,sdn-ovs,sync" and fluentd?

Comment (ravisantoshgudimetla) [reply roles/openshift_sdn/files/sdn-ovs.yaml]: While I agree with you, I think sizing of node should be done based on all the system-node critical and system-cluster critical pod. If node cannot hold all the pods then only, we should think of priority between "sdn,sdn-ovs,sync" and fluentd.

Comment (aveshagarwal) [reply roles/openshift_sdn/files/sdn-ovs.yaml]: > While I agree with you, I think sizing of node should be done based on all the system-node critical and system-cluster critical pod. If node cannot hold all the pods then only, we should think of priority between "sdn,sdn-ovs,sync" and fluentd.

If the sizing is like that the node can afford all "sdn,sdn-ovs,sync" and fluentd, why to worry about between cluster and node critical priorities?

Comment (ravisantoshgudimetla) [reply roles/openshift_sdn/files/sdn-ovs.yaml]: True as I told earlier, the chances of eviction are very less but this to be on safe side :)

Comment (sdodson) [reply roles/openshift_sdn/files/sdn-ovs.yaml]: Is there outstanding debate here?

**Comment by sdodson** [comment]
/approve
I'm not familiar with nuances of these priorities 
@ravisantoshgudimetla add / lgtm when you're happy

**Comment by ravisantoshgudimetla** [comment]
Thanks @sdodson - It was just a nit. I LGTM'ed.

**Comment by sdodson** [comment]
I assume we want to pick this back into release-3.10 or are these priorities only valid in 1.11?

**Comment by aveshagarwal** [comment]
> I assume we want to pick this back into release-3.10 or are these priorities only valid in 1.11?

Only 3.11.

**Pull request merged**